### PR TITLE
[performance] IFile.create: reduce 1 of 3 store.fetchInfo()

### PR DESCRIPTION
**Pull request by jukzi**
Assume the file does not exist (normal case) - otherwise implementation fails later during actual write.

https://github.com/eclipse-platform/eclipse.platform/issues/1443

### CODE REVIEW

File: resources/bundles/org.eclipse.core.resources/src/org/eclipse/core/internal/resources/File.java [hunk 1]
```diff
@@ -213,23 +214,13 @@ private void checkCreatable() throws CoreException {
...
-				if (localInfo.exists()) {
-					String name = getLocalManager().getLocalName(store);
-					if (name == null || localInfo.getName().equals(name)) {
-						delete(true, null);
-					} else {
-						// The file system is not case sensitive and there is already a file
-						// under this location.
-						message = NLS.bind(Messages.resources_existsLocalDifferentCase,
-								IPath.fromOSString(store.toString()).removeLastSegments(1).append(name).toOSString());
-						throw new ResourceException(IResourceStatus.CASE_VARIANT_EXISTS, getFullPath(), message, null);
```
[the remark this answers is not in the document]
Comment (jukzi) [reply resources/bundles/org.eclipse.core.resources/src/org/eclipse/core/internal/resources/File.java:228]: what exactly do you think is unused now?
image
image

**Comment by jukzi** [comment]
rebased. expecting error for 
".LocalOnlyFileHidden"	at org.eclipse.core.tests.resources.IFileTest.testCreate(IFileTest.java:464)

**Comment by jukzi** [comment]
As expected error on windows only:
```
failure in invocation 0 with inputs [L/OpenProject/ExistingFolder/.LocalOnlyFileHidden, java.io.ByteArrayInputStream@6ce7d6ab, true, org.eclipse.core.tests.harness.FussyProgressMonitor@53a3ccc]
invocation should succeed but unexpected exception occurred: org.eclipse.core.runtime.CoreException: Could not write file: D:\a\eclipse.platform\eclipse.platform\resources\tests\org.eclipse.core.tests.resources\target\work\data\OpenProject\ExistingFolder\.LocalOnlyFileHidden.
java.lang.AssertionError: 
failure in invocation 0 with inputs [L/OpenProject/ExistingFolder/.LocalOnlyFileHidden, java.io.ByteArrayInputStream@6ce7d6ab, true, org.eclipse.core.tests.harness.FussyProgressMonitor@53a3ccc]
invocation should succeed but unexpected exception occurred: org.eclipse.core.runtime.CoreException: Could not write file: D:\a\eclipse.platform\eclipse.platform\resources\tests\org.eclipse.core.tests.resources\target\work\data\OpenProject\ExistingFolder\.LocalOnlyFileHidden.
	at org.eclipse.core.tests.resources.TestPerformer.performTestRecursiveLoop(TestPerformer.java:96)
	at org.eclipse.core.tests.resources.TestPerformer.performTestRecursiveLoop(TestPerformer.java:111)
	at org.eclipse.core.tests.resources.TestPerformer.performTestRecursiveLoop(TestPerformer.java:111)
	at org.eclipse.core.tests.resources.TestPerformer.performTestRecursiveLoop(TestPerformer.java:111)
	at org.eclipse.core.tests.resources.TestPerformer.performTest(TestPerformer.java:58)
	at org.eclipse.core.tests.resources.IFileTest.testCreate(IFileTest.java:464)
Caused by: org.eclipse.core.runtime.CoreException: Could not write file: D:\a\eclipse.platform\eclipse.platform\resources\tests\org.eclipse.core.tests.resources\target\work\data\OpenProject\ExistingFolder\.LocalOnlyFileHidden.
	at org.eclipse.core.internal.filesystem.Policy.error(Policy.java:48)
	at org.eclipse.core.internal.filesystem.local.LocalFile.openOutputStream(LocalFile.java:507)
	at org.eclipse.core.internal.localstore.FileSystemResourceManager.write(FileSystemResourceManager.java:1227)
	at org.eclipse.core.internal.resources.File.internalSetContents(File.java:381)
	at org.eclipse.core.internal.resources.File.create(File.java:137)
	at org.eclipse.core.internal.resources.File.create(File.java:165)
	at org.eclipse.core.tests.resources.IFileTest$1.invokeMethod(IFileTest.java:430)
	at org.eclipse.core.tests.resources.TestPerformer.performTestRecursiveLoop(TestPerformer.java:90)
	... 97 more
Caused by: java.io.FileNotFoundException: D:\a\eclipse.platform\eclipse.platform\resources\tests\org.eclipse.core.tests.resources\target\work\data\OpenProject\ExistingFolder\.LocalOnlyFileHidden (Access is denied)
	at java.base/java.io.FileOutputStream.open0(Native Method)
	at java.base/java.io.FileOutputStream.open(FileOutputStream.java:289)
	at java.base/java.io.FileOutputStream.<init>(FileOutputStream.java:230)
	at org.eclipse.core.internal.filesystem.local.LocalFile.openOutputStream(LocalFile.java:497)
	... 103 more

```

**Comment by jukzi** [comment]
updated with fix for hidden files.

**Comment by jukzi** [comment]
if there are no objections i plan to merge tomorrow

**Comment by HannesWell** [comment]
Similar remark as in https://github.com/eclipse-platform/eclipse.platform/pull/1446#discussion_r1653653690.
Bottom line: We should be sure that this is a net improvement.

**Comment by jukzi** [comment]
> Similar remark as in [#1446 (comment)](https://github.com/eclipse-platform/eclipse.platform/pull/1446#discussion_r1653653690).
> Bottom line: We should be sure that this is a net improvement.

Same reasoning as there:  put a breakpoint in the catch clause and verify it never happens under normal conditions. It would only happen under the absurd conditions of:
* windows (likely)
* file is not up-to-date in workspace (only happens if resource API is not used)
* file is hidden (never set by eclipse ide)

**Pull request merged**